src/S3/Tests/Integration/S3ClientTest.php

```diff
@@ -486,9 +486,11 @@ public function testPutObjectAcl(): void
 
     public function testUploadFromClosure()
     {
-        $content = 'some content';
-        $closure = \Closure::fromCallable(function () use ($content) {
-            return $content;
```

Comment:
This is a change is the way I deal with callable.

Previously the callable returns the entiere body. Which is useless because everything will be set in memory.

I changed the behavior to use the same pattern that Curl and the HttpClient call the method until it returns an empty string.

An other solution could be to accept `\Generator` instead of callable.

Reply:
Could you document a proper callable in the S3 readme? (or somewhere you see fit)

Reply:
We should also document the argument to the callable

Reply:
Good Idea. 

Added a comment in the readme